managed/services/management/agent_test.go

```diff
@@ -0,0 +1,188 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package management
+
+import (
+	"context"
+	"testing"
+
+	"github.com/AlekSi/pointer"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/reform.v1"
+	"gopkg.in/reform.v1/dialects/postgresql"
+
+	"github.com/percona/pmm/api/managementpb"
+	"github.com/percona/pmm/managed/models"
+	"github.com/percona/pmm/managed/utils/logger"
+	"github.com/percona/pmm/managed/utils/testdb"
+	"github.com/percona/pmm/managed/utils/tests"
+)
+
+func setup(t *testing.T) (context.Context, *AgentService, func(t *testing.T), *mockPrometheusService) { //nolint:unparam
+	t.Helper()
+
+	ctx := logger.Set(context.Background(), t.Name())
+	uuid.SetRand(&tests.IDReader{})
+
+	sqlDB := testdb.Open(t, models.SetupFixtures, nil)
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+	vmdb := &mockPrometheusService{}
+	vmdb.Test(t)
+
+	state := &mockAgentsStateUpdater{}
+	state.Test(t)
+
+	ar := &mockAgentsRegistry{}
+	ar.Test(t)
+
+	teardown := func(t *testing.T) {
+		uuid.SetRand(nil)
+
+		require.NoError(t, sqlDB.Close())
+		vmdb.AssertExpectations(t)
+		state.AssertExpectations(t)
+		ar.AssertExpectations(t)
+	}
+	s := NewAgentService(db, ar)
+
+	return ctx, s, teardown, vmdb
+}
+
+func TestAgentService(t *testing.T) {
+	t.Run("List", func(t *testing.T) {
+		const (
+			pgExporterID      = "/agent_id/00000000-0000-4000-8000-000000000003"
+			pgStatStatementID = "/agent_id/00000000-0000-4000-8000-000000000004"
+		)
+
+		t.Run("Basic", func(t *testing.T) {
+			ctx, s, teardown, _ := setup(t)
+			defer teardown(t)
+
+			services, err := models.FindServices(s.db.Querier, models.ServiceFilters{})
+			require.NoError(t, err)
+			assert.Len(t, services, 1)
+			service := services[0]
+
+			s.r.(*mockAgentsRegistry).On("IsConnected", models.PMMServerAgentID).Return(true).Once() // PMM Server Agent
+			s.r.(*mockAgentsRegistry).On("IsConnected", pgExporterID).Return(false).Once()           // PMM Server PostgreSQL exporter
+			s.r.(*mockAgentsRegistry).On("IsConnected", pgStatStatementID).Return(false).Once()      // PMM Server PG Stat Statements agent
+			response, err := s.ListAgents(ctx, &managementpb.ListAgentRequest{
+				ServiceId: service.ServiceID,
+			})
+
+			require.NoError(t, err)
+			assert.Len(t, response.Agents, 3) // 2 exporters + 1 agent
```

Comment:
should we check that the correct agents are returned?

Reply:
Done.